src/main/java/org/myrobotlab/service/config/SpeechRecognizerConfig.java

```diff
@@ -1,9 +1,18 @@
 package org.myrobotlab.service.config;
 
+import org.myrobotlab.framework.Plan;
+
 public class SpeechRecognizerConfig extends ServiceConfig {
 
-  public boolean listening;
+  public boolean listening = false;
+  public boolean recording = false;
```

Comment:
should a speech recognizer be able to record?  I'm not sure this should be in the part of the base config class here.  speech recognizers that handle the audio outside of mrl won't be able to do this i assume/guess.

Reply:
updated javadocs
```java
 /**
   * status of listening, the robot will publish all recognized text
   */
  protected boolean isListening = false;

  /**
   * Recording means that data is being sent to the speech recognizer,
   * however, it will not be acted on it unless its listening.  Recording = true
   * and listening=false is the common state for a wake word trigger the robot
   * into listening=true state.  If recording is off, there is no data sent
   * to the speech recognizer
   */
  protected boolean isRecording = false;
  ```